File: Cap 12/Topico 12.7/Polimorfismo/Universal/Inclusao/polimuninc.scala
/*
 * Livro .....: Algoritmos
 *              Lógica para Desenvolvimento de Programação de
 *              Computadores
 * Autores ...: José Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Érica
 *
 * Capítulo ..: 12 - Tópico 12.7
 * 
 * Programa ..: POLIFORMISMO UNIVERSAL DE INCLUSÃO
 * 
 */
 
abstract class CLS_PAI {
  def EXECUTA(): Unit = {
    println("Ação executada na classe-pai")
  }
}

class CLS_FILHO extends CLS_PAI {
  override def EXECUTA(): Unit = {
    println("Ação executada na classe-filho")
  }
}

object Main extends App {
	
  // Classe anônima que herda de CLS_PAI	
  var PAI: CLS_PAI = new CLS_PAI { } 
  var FILHO: CLS_FILHO = new CLS_FILHO()

  // PAI.EXECUTA()
  PAI.EXECUTA()

  // FILHO.EXECUTA()
  FILHO.EXECUTA()

  // PAI <- FILHO
  PAI = FILHO

  // PAI.EXECUTA()
  PAI.EXECUTA()

  // PAI <- ""
  PAI = new CLS_PAI { } // Classe anônima que herda de CLS_PAI

  // PAI.EXECUTA()
  PAI.EXECUTA()
}

File: Cap 12/12.7 - Poliformismo/12.7.1 - Poliformismo Universal de Inclusao/polimuninc.php
<?php

/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.7
 * 
 * Programa ..: POLIFORMISMO UNIVERSAL DE INCLUSAO
 * 
 */

class CLS_PAI 
{
  public function EXECUTA() 
  {
    echo "Acao executada na classe-pai\n";
  }
}

class CLS_FILHO extends CLS_PAI 
{
  public function EXECUTA() 
  {
    echo "Acao executada na classe-filho\n";
  }
}

$PAI = new CLS_PAI();
$FILHO = new CLS_FILHO();
$ptrPAI;

$PAI->EXECUTA();

$FILHO->EXECUTA();

$ptrPAI = $FILHO; // PAI ← FILHO

$ptrPAI->EXECUTA();

$ptrPAI = $PAI; // PAI ← ""

$PAI->EXECUTA();

?>
